stdlib/InteractiveUtils/src/macros.jl

```diff
@@ -83,6 +83,15 @@ macro which(ex0::Symbol)
     return :(which($__module__, $ex0))
 end
 
+macro code_typed(optimize, ex0)
+    thecall = gen_call_with_extracted_types(__module__, :code_typed, ex0)
+    insert!(thecall.args, 2, :($(esc(Expr(:parameters, :(optimize=$optimize))))))
+    quote
+        results = $thecall
+        length(results) == 1 ? results[1] : results
+    end
+end
```

Comment:
Perhaps remove the eval-loop below instead, and define `code_typed` and `code_lowered` separately, with the kwarg for `code_typed`?